src/inmanta/plugins.py

```diff
@@ -191,21 +191,136 @@ def clear(cls, inmanta_module: Optional[str] = None) -> None:
             cls.__functions = {}
 
 
-class PluginArgument:
+class Null(inmanta_type.Type):
+    """
+    This custom type is used for the validation of plugins which only
+    accept null as an argument or return value.  It is meant to be used
+    as the element_type of an instance of the `NullableType` class.
+    """
+
+    def validate(self, value: object | None) -> bool:
+        return isinstance(value, NoneValue)
+
+    def type_string(self) -> str | None:
```

Comment:
I copy-pasted the signature from the parent class, but indeed, this one is always returning a string.  Is it considered good practice to always "retrain" the return type of a sub class's method when we can ensure that the returned type will indeed never be one of the parent's offerings?

Reply:
I wouldn't call it "retrain", more "refine" because you still have to be compatible with the parent. i.e. you can return something more specific, but not less so. And indeed I typically try to do this whenever possible, unless there's a chance the signature might change due to a change of implementation. Afaik it is indeed considered good practice.